src/im/InformationManager.cc

```diff
@@ -187,6 +172,31 @@ void InformationManager::delete_host(int hid)
 /* -------------------------------------------------------------------------- */
 /* -------------------------------------------------------------------------- */
 
+void InformationManager::raft_status(bool leader)
+{
+    if (leader)
+    {
+        send_host_pool();
```

Comment:
send_host_pool is only used here, better put the code here and reuse the get_dirver function call and error handling